unlock-app/src/components/content/certification/CertificationForm.tsx

```diff
@@ -0,0 +1,371 @@
+  ToggleSwitch,
+  ImageUpload,
+  Select,
+} from '@unlock-protocol/ui'
+import { useConfig } from '~/utils/withConfig'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { networkDescription } from '~/components/interface/locks/Create/elements/CreateLockForm'
+import { useQuery } from '@tanstack/react-query'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { SelectCurrencyModal } from '~/components/interface/locks/Create/modals/SelectCurrencyModal'
+import { UNLIMITED_KEYS_DURATION } from '~/constants'
+import { CryptoIcon } from '@unlock-protocol/crypto-icon'
+import { useImageUpload } from '~/hooks/useImageUpload'
+import { BalanceWarning } from '~/components/interface/locks/Create/elements/BalanceWarning'
+// TODO replace with zod, but only once we have replaced Lock and MetadataFormData as well
+export interface NewEventForm {
+  network: number
+  lock: Omit<Lock, 'address' | 'key'>
+  currencySymbol: string
+  metadata: Partial<MetadataFormData>
+}
+
+interface FormProps {
+  onSubmit: (data: NewEventForm) => void
+}
+
+export const CertificationForm = ({ onSubmit }: FormProps) => {
+  const { networks } = useConfig()
+  const { network, account } = useAuth()
+
+  const [isFree, setIsFree] = useState(true)
+  const [unlimitedCapacity, setUnlimitedCapacity] = useState(false)
+  const [forever, setForever] = useState(false)
+  const [isCurrencyModalOpen, setCurrencyModalOpen] = useState(false)
+  const { mutateAsync: uploadImage, isLoading: isUploading } = useImageUpload()
+
+  const web3Service = useWeb3Service()
+
+  const methods = useForm<NewEventForm>({
+    mode: 'onChange',
+    shouldUnregister: false,
+    defaultValues: {
+      network,
+      lock: {
+        name: '',
+        expirationDuration: UNLIMITED_KEYS_DURATION,
+        maxNumberOfKeys: 100,
+        currencyContractAddress: null,
+        keyPrice: '0',
+      },
+      currencySymbol: networks[network!].nativeCurrency.symbol,
+      metadata: {
+        description: '',
+        certification: {
+          issuer: '',
+          expire_date: null,
+        },
+        image: '',
+      },
+    },
+  })
+
+  const {
+    control,
+    register,
+    setValue,
+    formState: { errors },
+    watch,
+  } = methods
+
+  const details = useWatch({
+    control,
+  })
+
+  const DescDescription = () => (
+    <p>
+      Give detail about this certificate to increase the credibility.{' '}
+      <a
+        className="text-brand-ui-primary hover:underline"
+        target="_blank"
+        rel="noopener noreferrer"
+        href="https://www.markdownguide.org/cheat-sheet"
+      >
+        Markdown is supported.
+      </a>
+    </p>
+  )
+
+  const { isLoading: isLoadingBalance, data: balance } = useQuery(
+    ['getBalance', account, details.network],
+    async () => {
+      if (!details.network) {
+        return 1.0
+      }
+      return parseFloat(
+        await web3Service.getAddressBalance(account!, details.network!)
```

Comment:
What is this???

Reply:
Also if you are just trying the balance we already have a hook for this

Reply:
Get balance to eventually show this message: 

```
You currently do not have any ${currency} token to pay for gas to deploy on the ${networkName} network.`
```